### ISSUE TITLE
Gradient descent or stochastic gradient descent

### ISSUE THREAD

**Issue by utkarsh0902311047** [SPFlow/SPFlow#131]
### Describe your request

Please share a tutorial example on performing gradient descent or stochastic gradient descent tests to update weights for generative and discriminative SPN.
Also, if the input data is univariate or multivariate does the mean and variance get updated too along with the weights 

### Briefly explain its use-case

It will help to know insights into the training of an SPN

**Comment by braun-steven** [comment]
Hi, we will provide tutorials in the upcoming 1.0.0 version of SPFlow which is currently in development. 

Until then, you can check out the [`optimize_tf`](https://github.com/SPFlow/SPFlow/blob/be2828c6e79ccd3b4c0bde6717ca3f58fc06458a/src/spn/gpu/TensorFlow.py#L109) method, which takes an SPN object and performs gradient descent using TensorFlow to minimize the sum of negative log-likelihoods (NLL loss).

https://github.com/SPFlow/SPFlow/blob/be2828c6e79ccd3b4c0bde6717ca3f58fc06458a/src/spn/gpu/TensorFlow.py#L109
